micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/config/PropertyAbsentOrEnabledCondition.groovy

```diff
@@ -0,0 +1,31 @@
+package com.ofg.infrastructure.config
+
+import groovy.transform.CompileStatic
+import org.apache.commons.lang.BooleanUtils
+import org.springframework.context.annotation.Condition
+import org.springframework.context.annotation.ConditionContext
+import org.springframework.core.type.AnnotatedTypeMetadata
+
+@CompileStatic
+abstract class PropertyAbsentOrEnabledCondition implements Condition {
+
+    private final String filteringProperty
+
+    PropertyAbsentOrEnabledCondition(String filteringProperty) {
+        this.filteringProperty = filteringProperty
+    }
+
+    @Override
+    boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+        return switchNotDefined(context) || switchIsEnabled(context)
+    }
+
+    private boolean switchNotDefined(ConditionContext context) {
+        return !context.environment.containsProperty(filteringProperty)
+    }
+
+    private boolean switchIsEnabled(ConditionContext context) {
+        String propertyEnabled = context.environment.getProperty(filteringProperty).toLowerCase()
```

Comment:
In one case you have getProperty() and in the other you have contains() we would have to in both cases do a get...() and once use it and in the other case check it against null.